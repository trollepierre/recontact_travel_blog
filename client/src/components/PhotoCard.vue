<template xmlns:v-lazy="http://www.w3.org/1999/xhtml">
  <div class="photo-card">
    <article class="photo">
      <div class="photo__content">
        <div ref="container">
          <img
            v-lazy="imgLink"
            :alt="image">
          <div v-lazy:background-image="imgLink"/>
        </div>
      </div>
    </article>
  </div>
</template>

<script>
  export default {
    name: 'PhotoCard',
    props: {
      photo: { type: Object, default: () => {} },
    },
    computed: {
      imgLink() {
        const { imgLink } = this.photo
        return !imgLink ? false : imgLink
      },
      image() {
        return this.$t('alt')
      },
    },
    mounted() {
      this.$Lazyload.lazyLoadHandler()
    },
    i18n: {
      messages: {
        fr: {
          alt: 'Une image',
        },
        en: {
          alt: 'An image',
        },
      },
    },
  }
</script>

<style scoped>
  img {
    max-width: 100%;
    max-height: 40vw;
  }

  @media only screen and (max-width: 640px) {
    img {
      max-height: 100%;
    }
  }

  .photo {
    min-width: 260px;
    background: #ffffff;
    border-radius: 4px !important;
    box-shadow: 0 1px 1px rgba(0, 0, 0, .15);
    border: 1px solid rgba(0, 0, 0, .09);
    display: flex;
    flex-direction: column;
    color: #535a60;
  }

  .photo__content {
    font-size: 15px;
    padding: 15px;
    display: block;
    color: #000;
    text-align: center;
  }

</style>
