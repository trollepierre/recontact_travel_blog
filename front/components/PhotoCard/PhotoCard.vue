<template xmlns:v-lazy="http://www.w3.org/1999/xhtml">
  <article class="photo">
    <div
      ref="container"
      class="photo__content">
      <img
        v-lazy="imgLink"
        class="img"
        rel="noreferrer"
        :alt="image">
      <div v-lazy:background-image="imgLink"/>
    </div>
  </article>
</template>

<script>
  export default {
    name: 'PhotoCard',
    props: {
      photo: {
        type: Object,
        default: () => {
        },
      },
    },
    computed: {
      imgLink() {
        const { imgLink } = this.photo
        return !imgLink ? false : imgLink
      },
      image() {
        return this.$t('alt')
      },
    },
    mounted() {
      this.$Lazyload.lazyLoadHandler()
    },
    i18n: {
      messages: {
        fr: {
          alt: 'Une image',
        },
        en: {
          alt: 'An image',
        },
      },
    },
  }
</script>

<style  lang="scss" scoped>
  .img {
    max-width: 100%;
    max-height: 40vw;
  }

  @media only screen and (max-width: 640px) {
    .img {
      max-height: 100%;
    }
  }

  .photo {
    min-width: 260px;
    background: $card-bg;
    border-radius: 4px !important;
    box-shadow: 0 1px 1px $border;
    border: 1px solid $border;
    display: flex;
    flex-direction: column;
  }

  .photo__content {
    font-size: 15px;
    padding: 15px;
    display: block;
    color: $card-placeholder;
    text-align: center;
  }

</style>
