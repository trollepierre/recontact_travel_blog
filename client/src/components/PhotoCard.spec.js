import Vue from 'vue';
import PhotoCard from './PhotoCard';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import VueLazyload from 'vue-lazyload';

describe('Component | PhotoCard.vue', () => {
  let wrapper
  let photo;
  let localVue;

  beforeEach(() => {
    photo = {
      imgLink: 'webf',
      dropboxId: '45',
    };
    localVue = createLocalVue();
    localVue.use(VueI18n)
    localVue.use(VueRouter)
    localVue.use(VueLazyload)
    const propsData = { photo };
    wrapper = shallowMount(PhotoCard, { localVue, propsData })
  });

  it('should be named "PhotoCard"', () => {
    expect(wrapper.name()).toEqual('PhotoCard');
  });

  describe('template', () => {
    it('should match snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  xdescribe('render', () => {
    it('should render photo image', () => Vue.nextTick().then(() => {
      const photoLink = wrapper.find('img');
      expect(photoLink.getAttribute('src')).toEqual('data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7');
      expect(photoLink.getAttribute('lazy')).toEqual('loading');
    }));

    // Comment tester le lazy load ?
    it.skip('should render photo image', () => Vue.nextTick().then(() => {
      const photoLink = wrapper.find('img');

      // return Vue.nextTick().then(() => {
      // console.log(photoLink);
      expect(photoLink.getAttribute('lazy')).toEqual('loaded');
      expect(photoLink.getAttribute('src')).toContain('webf');
      // });
    }));
  });

  xdescribe('computed property #imgLink', () => {
    it('should return imgLink when defined', () => {

      photo.imgLink = 'dropbox.com/img0.jpg';


      const { imgLink } = component;


      expect(imgLink).toEqual('dropbox.com/img0.jpg');
    });

    it('should return false when api status is undefined', () => {

      photo.imgLink = '';


      const { imgLink } = component;


      expect(imgLink).toEqual(false);
    });
  });
});
