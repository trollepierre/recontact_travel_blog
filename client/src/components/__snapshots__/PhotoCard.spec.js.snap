// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | PhotoCard.vue template should match snapshot 1`] = `
<div
  class="photo-card"
>
  <article
    class="photo"
  >
    <div
      class="photo__content"
    >
      <div>
        <img
          alt="alt"
          data-src="webf"
          lazy="loading"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
         
        <div
          data-src="webf"
          lazy="loading"
          style="background-image: url(data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7);"
        />
      </div>
    </div>
  </article>
</div>
`;
